### PR TITLE
Adds 3 new hats + extra mime beret colours

### PR DESCRIPTION
**Pull request by WALPVRGIS**
[SPRITES] [WIKI]

## About the PR 

- Adds baseball caps in 8 colours; these can be used inhand to flip them backwards (so you can be radicool).
- Adds costume goggles in 5 colours; these count as hats and can only be worn on your head, for that good space mechanic aesthetic.
- Adds pirate hats in 2 colours; I just wanted these for spooktober aaaa. 👻 
- Adds some recolours of the mime beret, named French berets.
- Moves hairclips to the accessory section in the clothing booth as opposed to headwear (made sense to me).

hats1

Sorry if my baseball cap flipping code is stinky. :( Feel free to critique/offer suggestions for better methods.

## Why's this needed? 

- Character customization 😸 
- Spooktober!

## Changelog

```changelog
(u)Walpvrgis
(*)Added a few new hats in multiple colours; baseball caps, goggles (worn as a headband) and pirate hats, as well as some extra mime beret colours. These can be found in the clothing booth under Accessories and Headwear.
```

### CODE REVIEW

File: code/obj/item/clothing/hats.dm [hunk 1]
```diff
@@ -1611,3 +1611,176 @@ ABSTRACT_TYPE(/obj/item/clothing/head/hairbow)
...
+// Baseball Caps
+
+ABSTRACT_TYPE(/obj/item/clothing/head/basecap)
+/obj/item/clothing/head/basecap
+	name = "baseball cap"
+	desc = "Wear it normally, or flip it backwards to increase your coolness."
+	uses_multiple_icon_states = 1
+	icon = 'icons/obj/clothing/item_hats.dmi'
+	wear_image_icon = 'icons/mob/head.dmi'
+	var/hatflip = 1
```
Comment (Zonespace27) [inline code/obj/item/clothing/hats.dm:1710]:
```suggestion
	var/hatflip = TRUE
```

File: code/obj/item/clothing/hats.dm [hunk 2]
```diff
@@ -1611,3 +1611,176 @@ ABSTRACT_TYPE(/obj/item/clothing/head/hairbow)
...
+
+	New()
+		..()
+		name = "[hatcolour] baseball cap"
+		item_state = "basecap_[hatcolour]"
+
+	attack_self(var/mob/user as mob)
+		if(user.r_hand == src || user.l_hand == src)
+			if(!src.hatflip)
+				src.hatflip = 1
```
Comment (Zonespace27) [inline code/obj/item/clothing/hats.dm:1721]:
```suggestion
				src.hatflip = TRUE
```

File: code/obj/item/clothing/hats.dm [hunk 3]
```diff
@@ -1611,3 +1611,176 @@ ABSTRACT_TYPE(/obj/item/clothing/head/hairbow)
...
+				src.icon_state = "basecap_[hatcolour]"
+				src.item_state = "basecap_[hatcolour]"
+				boutput(user, "<span class='notice'>You flip your baseball cap back into the standard baseball cap position.</span>")
+			else
+				src.hatflip = 0
```
Comment (Zonespace27) [inline code/obj/item/clothing/hats.dm:1726]:
```suggestion
				src.hatflip = FALSE
```

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: code/obj/item/clothing/hats.dm (hunk 3 above)
Comment (WALPVRGIS) [reply code/obj/item/clothing/hats.dm:1726]: Prayer hands emoji thank you!!

File: code/obj/item/clothing/hats.dm [hunk 4]
```diff
@@ -1611,3 +1611,176 @@ ABSTRACT_TYPE(/obj/item/clothing/head/hairbow)
...
+	var/hatflip = TRUE
+	var/hatcolour = "black"
+
+	New()
+		..()
+		name = "[hatcolour] baseball cap"
+		item_state = "basecap_[hatcolour]"
+
+	attack_self(var/mob/user as mob)
+		if(user.r_hand == src || user.l_hand == src)
```
Comment (Zonespace27) [inline code/obj/item/clothing/hats.dm:1719]:
```suggestion
		
```
This is probably redundant but I'm reviewing this on mobile so I can't guarantee that it's unneeded.

File: code/obj/item/clothing/hats.dm [hunk 5]
```diff
@@ -1611,3 +1611,176 @@ ABSTRACT_TYPE(/obj/item/clothing/head/hairbow)
...
+				src.hatflip = TRUE
+				src.icon_state = "basecap_[hatcolour]"
+				src.item_state = "basecap_[hatcolour]"
+				boutput(user, "<span class='notice'>You flip your baseball cap back into the standard baseball cap position.</span>")
+			else
+				src.hatflip = FALSE
+				src.icon_state = "basecapflip_[hatcolour]"
+				src.item_state = "basecapflip_[hatcolour]"
+				boutput(user, "<span class='notice'>You flip your baseball cap around. Now it's backwards.</span>")
+			return
```
Comment (Zonespace27) [inline code/obj/item/clothing/hats.dm:1730]:
```suggestion
			
```
Unneeded return

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: code/obj/item/clothing/hats.dm (hunk 5 above)
Comment (WALPVRGIS) [reply code/obj/item/clothing/hats.dm]: Thanks again! :D

**Review by Sovexe — CHANGES_REQUESTED** [review]

File: code/modules/vending/clothingbooth_items.dm [hunk 6]
```diff
@@ -785,6 +818,77 @@ ABSTRACT_TYPE(/datum/clothingbooth_item/head)
 	path = /obj/item/clothing/head/link
 	cost = PAY_TRADESMAN
 
+ABSTRACT_TYPE(/datum/clothingbooth_item/head/frenchberet)
+/datum/clothingbooth_item/head/frenchberet
+	name = "Baseball Cap"
```
[the remark this answers is not in the document]
Comment (WALPVRGIS) [reply code/modules/vending/clothingbooth_items.dm:823]: LOL OH NO I can't believe I've done this 😔 🤣

File: code/obj/item/clothing/hats.dm [hunk 7]
```diff
@@ -1611,3 +1611,174 @@ ABSTRACT_TYPE(/obj/item/clothing/head/hairbow)
...
+		if(!src.hatflip)
+			src.hatflip = TRUE
+			src.icon_state = "basecap_[hatcolour]"
+			src.item_state = "basecap_[hatcolour]"
+			boutput(user, "<span class='notice'>You flip your baseball cap back into the standard baseball cap position.</span>")
+		else
+			src.hatflip = FALSE
+			src.icon_state = "basecapflip_[hatcolour]"
+			src.item_state = "basecapflip_[hatcolour]"
+			boutput(user, "<span class='notice'>You flip your baseball cap around. Now it's backwards.</span>")
```
[the remark this answers is not in the document]
Comment (WALPVRGIS) [reply code/obj/item/clothing/hats.dm]: Made this change, but kept the `"basecapflip_[hatcolour]"`s just to keep the item/icon state swap working. :)

**Pull request merged**